pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -864,14 +865,51 @@ def __init__(
 
         self._target_wire = wires[0]
         self._work_wires = work_wires
+        self._control_wires = control_wires
+
+        U = PauliX.matrix
+        self.U = U
```

Comment:
I don't think we need to set the `self.U` attribute anymore, as it was fairly specific to the `ControlledQubitUnitary` gate. Anytime we need that parameter, we could instead just provide the matrix itself.

Reply:
Yes! That is a better option.